libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m

```diff
@@ -51,63 +52,48 @@
 NSString * const kPathsArg            = @"paths";
 NSString * const kReIndexDataArg      = @"reIndexData";
 NSString * const kIsGlobalStoreArg    = @"isGlobalStore";
+NSString * const kStoreName           = @"storeName";
 
 @interface SFSmartStorePlugin() 
 
-@property (nonatomic, strong) SFSmartStoreInspectorViewController *inspector;
-@property (nonatomic, strong) SFSmartStoreInspectorViewController *globalInspector;
-@property (nonatomic, strong) SFSmartStore *store;
-@property (nonatomic, strong) SFSmartStore *globalStore;
-@property (nonatomic, strong) NSMutableDictionary *userCursorCache;
-@property (nonatomic, strong) NSMutableDictionary *globalCursorCache;
-
+@property (nonatomic, strong) SFMutableDictionaryThreadSafe *databaseInspectors;
+@property (nonatomic, strong) SFMutableDictionaryThreadSafe *cursorCache;
 @end
 
 @implementation SFSmartStorePlugin
 
 - (void)resetCursorCaches
 {
-    [[self userCursorCache] removeAllObjects];
-    [[self globalCursorCache] removeAllObjects];
-}
-
-- (SFSmartStore *)store
-{
-    return [SFSmartStore sharedStoreWithName:kDefaultSmartStoreName];
-}
-
-- (SFSmartStore *)globalStore
-{
-    return [SFSmartStore sharedGlobalStoreWithName:kDefaultSmartStoreName];
+    [[self cursorCache] removeAllObjects];
 }
 
 - (void)pluginInitialize
 {
     [self log:SFLogLevelDebug msg:@"SFSmartStorePlugin pluginInitialize"];
-    self.userCursorCache = [[NSMutableDictionary alloc] init];
-    self.globalCursorCache = [[NSMutableDictionary alloc] init];
-    self.inspector = [[SFSmartStoreInspectorViewController alloc] initWithStore:self.store];
-    self.globalInspector = [[SFSmartStoreInspectorViewController alloc] initWithStore:self.globalStore];
+    self.cursorCache = [[SFMutableDictionaryThreadSafe alloc] init];
+    self.databaseInspectors = [[SFMutableDictionaryThreadSafe alloc] init];
 }
 
 #pragma mark - Object bridging helpers
+- (SFStoreCursor*)cursorByCursorId:(NSString*)cursorId isGlobal:(BOOL)isGlobal {
```

Comment:
I'm having a hard time articulating what I want to be different here, specifically.  Generally, this cursor ID pattern is hard to follow, which is going to make it hard to maintain.

My suggestion is to pull the internal cursor ID stuff out into its own object, and then consolidate the accessor functionality in this class accordingly.  There are simply too many methods and parameters flying around here, that do nearly the same thing.  It's hard to follow.